kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -103,7 +100,7 @@ class SparseMatrixMultiplicationUtility
     /// Metafunction that returns value type of a matrix or a vector type.
     template <class T, class Enable = void>
     struct value_type {
-        typedef typename T::value_type type;
+        using type = typename T::value_type;
     };
```

Comment:
I don't get what this nested `value_type` does that `typename T::value_type` cannot. Is it possible that this is just some leftover that's no longer necessary? Can you please try replacing it?